app/src/main/java/fr/free/nrw/commons/di/ApplicationlessInjection.java

```diff
@@ -27,7 +27,7 @@
         HasBroadcastReceiverInjector,
         HasContentProviderInjector {
 
-    private static ApplicationlessInjection instance = null;
+    private static volatile ApplicationlessInjection instance = null;
```

Comment:
Necessary? Feel this is more than a lint issue.

Reply:
Double checked locking can result in insidious bugs while being used by multiple threads. Please refer the following link for a detailed explanation.

https://en.wikipedia.org/wiki/Double-checked_locking#Usage_in_Java

Reply:
I don't think @domdomegg was saying that this change is wrong, just that it should be separated into a different PR. :) When we modularize PRs, it is easier to check which PR caused a specific bug, and to revert if necessary.

Reply:
Aha. That makes sense. I'll group up and push stuff like this in another PR then.